core/docker_service.go

```diff
@@ -326,14 +340,37 @@ func (ds *dockerService) AlphaVersion(
 }
 
 // getDockerVersion gets the version information from docker.
-func (ds *dockerService) getDockerVersion() (*dockertypes.Version, error) {
-	v, err := ds.client.Version()
+func (ds *dockerService) getDockerVersion() (v *dockertypes.Version, err error) {
+	if ds.versionCache != nil {
+		v, err = ds.getDockerVersionFromCache()
+	} else {
+		v, err = ds.client.Version()
+		fixAPIVersion(v)
+	}
 	if err != nil {
 		return nil, fmt.Errorf("failed to get docker version: %v", err)
 	}
-	// Docker API version (e.g., 1.23) is not semver compatible. Add a ".0"
-	// suffix to remedy this.
-	v.APIVersion = fmt.Sprintf("%s.0", v.APIVersion)
+	return v, nil
+}
+
+// fixAPIVersion remedy Docker API version (e.g., 1.23) which is not semver compatible by
+// adding a ".0" suffix
+func fixAPIVersion(v *dockertypes.Version) {
```

Comment:
This seems like something we should handle on our end. Do we know the scope of anyone else using the version value? 

Even though I agree semvar should be used, it's not being used. This looks like there is a docker api version vx.x.0 and if you go to the docker page, there isn't one. I think this could lead to a lot of confusion.

Reply:
@nwneisen - what does "our end" mean?

Reply:
@vikramhh Modify the version to be semvar where we were having an issue in company code rather than in the community `cri-dockerd` code that effects multiple parties

Reply:
Yes, it looks kube CRI interface does not need docker API version.

Reply:
So the version isn't being used anywhere (no bug to fix) and we are adding a log line to report a non existent version of the docker api is being used?

Reply:
So not being semvar isn't causing us issues (no bug to fix) but we are changing `cri-dockerd` to report a non existent version of the docker api?

Reply:
I see that this is already being done. I'd say leave it as is then

Reply:
This PR does not make any functional changes. cri-dockerd still reports the same API version as before.